src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/transformers/QuadFaceStripper.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2018 covers1624
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package appeng.thirdparty.codechicken.lib.model.pipeline.transformers;
+
+
+import net.minecraft.util.EnumFacing.AxisDirection;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraftforge.client.model.pipeline.IVertexConsumer;
+
+import appeng.thirdparty.codechicken.lib.model.Quad.Vertex;
+import appeng.thirdparty.codechicken.lib.model.pipeline.IPipelineElementFactory;
+import appeng.thirdparty.codechicken.lib.model.pipeline.QuadTransformer;
+
+import static net.minecraft.util.EnumFacing.AxisDirection.POSITIVE;
+
+
+/**
+ * This transformer strips quads that are on faces.
+ * Simply set the bounds for the faces, and the strip mask.
+ *
+ * @author covers1624
+ */
+public class QuadFaceStripper extends QuadTransformer
+{
+
+	public static final IPipelineElementFactory<QuadFaceStripper> FACTORY = QuadFaceStripper::new;
+
+	private AxisAlignedBB bounds;
+	private int mask;
+
+	QuadFaceStripper()
+	{
+		super();
+	}
+
+	public QuadFaceStripper( IVertexConsumer parent, AxisAlignedBB bounds, int mask )
+	{
+		super( parent );
+		this.bounds = bounds;
+		this.mask = mask;
+	}
+
+	/**
+	 * The bounds of the faces,
+	 * used as the .. bounds, if all vertices of a quad
+	 * lay on the bounds, it is up for stripping.
+	 *
+	 * @param bounds The bounds.
+	 */
+	public void setBounds( AxisAlignedBB bounds )
+	{
+		this.bounds = bounds;
+	}
+
+	/**
+	 * The mask to strip edges.
+	 * This is an opt in system,
+	 * the mask is simple 'mask = (1 << side)'.
+	 *
+	 * @param mask The mask.
+	 */
+	public void setMask( int mask )
+	{
+		this.mask = mask;
+	}
+
+	@Override
+	public boolean transform()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "transform" is 18 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)